js/client.js

```diff
@@ -709,7 +709,17 @@ function toId() {
 		connect: function () {
 			if (this.down) return;
 
-			if (Config.server.banned || (Config.bannedHosts && Config.bannedHosts.indexOf(Config.server.host) >= 0)) {
+			var serverIsBanned = false;
+			if (Config.bannedHosts) {
+				for (var i = 0; i < Config.bannedHosts.length; i++) {
+					var host = Config.bannedHosts[i];
+					if (Config.server.host.includes(host)) {
+						serverIsBanned = true;
+					}
+				}
+			}
+
+			if (Config.server.banned || serverIsBanned) {
```

Comment:
```suggestion
			if (Config.bannedHosts) {
				for (var i = 0; i < Config.bannedHosts.length; i++) {
					var host = Config.bannedHosts[i];
					if (Config.server.host.includes(host)) {
						Config.server.banned = true;
						break;
					}
				}
			}

			if (Config.server.banned) {
```

There's no real need for the intermediate variable, either.